user-service/src/main/java/hcmus/alumni/userservice/controller/UserServiceController.java

```diff
@@ -484,12 +513,602 @@ public ResponseEntity<HashMap<String, Object>> getSearchResult(
+			throw new AppException(22001, "Không tìm thấy công việc", HttpStatus.NOT_FOUND);
+		}
+
+		JobModel jobToUpdate = optionalJob.get();
+
+		Optional<JobModel> existingJob = jobRepository.findByUserIdAndCompanyNameAndPosition(id,
+				updatedJobDto.getCompanyName(), updatedJobDto.getPosition());
+		if (existingJob.isPresent() && !existingJob.get().getJobId().equals(jobId)) {
+			throw new AppException(22002, "Không thể cập nhật vì đã tồn tại công việc với cùng công ty và chức vụ",
+					HttpStatus.BAD_REQUEST);
+		}
+
+		if (updatedJobDto.getCompanyName() != null || !updatedJobDto.getCompanyName().isEmpty()) {
+			jobToUpdate.setCompanyName(updatedJobDto.getCompanyName());
+		}
+		if (updatedJobDto.getPosition() != null && !updatedJobDto.getPosition().isEmpty()) {
+			jobToUpdate.setPosition(updatedJobDto.getPosition());
+		}
+		if (updatedJobDto.getStartTime() != null) {
+			jobToUpdate.setStartTime(updatedJobDto.getStartTime());
+		}
+		if (updatedJobDto.getEndTime() != null) {
+			jobToUpdate.setEndTime(updatedJobDto.getEndTime());
+		}
+		if (updatedJobDto.getPrivacy() != null) {
+			jobToUpdate.setPrivacy(Privacy.valueOf(updatedJobDto.getPrivacy().toUpperCase()));
+		}
+		if (updatedJobDto.getIsWorking() != null) {
+			jobToUpdate.setIsWorking(updatedJobDto.getIsWorking());
+		}
+		if (updatedJobDto.getIsDelete() != null) {
+			jobToUpdate.setIsDelete(updatedJobDto.getIsDelete());
+		}
+
+		jobRepository.save(jobToUpdate);
+
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+
+	@GetMapping("profile/{id}/education")
+	public ResponseEntity<Map<String, Object>> getAllEducations(@PathVariable String id) {
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+		if (optionalUser.isEmpty()) {
+			throw new AppException(22101, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+
+		List<EducationModel> edus = educationRepository.findByUserId(id);
+		if (edus.isEmpty()) {
+			throw new AppException(22102, "Không có bằng cấp nào được tìm thấy cho người dùng này",
+					HttpStatus.NOT_FOUND);
+		}
+
+		HashMap<String, Object> result = new HashMap<>();
+		result.put("edus", edus);
+		return ResponseEntity.status(HttpStatus.OK).body(result);
+	}
+
+	@GetMapping("profile/{id}/education/{educationId}")
+	public ResponseEntity<IEducationDto> getEducationById(@PathVariable String id, @PathVariable String educationId) {
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+		if (optionalUser.isEmpty()) {
+			throw new AppException(22201, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+
+		Optional<IEducationDto> optionalEdu = educationRepository.findByEducationId(educationId);
+		if (optionalEdu.isEmpty()) {
+			throw new AppException(22202, "Không có bằng cấp nào được tìm thấy cho người dùng này",
+					HttpStatus.NOT_FOUND);
+		}
+		return ResponseEntity.status(HttpStatus.OK).body(optionalEdu.get());
+	}
+
+	@PostMapping("profile/{id}/create-education")
```

Comment:
```suggestion
	@PostMapping("/profile/education")
```
lý do như api post job, avatar, cover